src/commands/moderators/report.ts

```diff
@@ -22,24 +30,62 @@ export default defineCommandHandler({
     const reporterId = interaction.user.id
     const reporteeId = member.id
 
-    // TODO: Maybe allow a reason to be specified? e.g. by using a modal
-    const reason = 'Reported via context menu'
+    const reasonModal = new ModalBuilder()
+      .setCustomId('reasonModal')
+      .setTitle('Report to moderator')
+
+    const reasonInput = new TextInputBuilder()
+      .setCustomId('reasonInput')
+      .setLabel('Reason')
+      .setStyle(TextInputStyle.Paragraph)
+      .setRequired(true)
+
+    reasonModal.addComponents(
+      new ActionRowBuilder<ModalActionRowComponentBuilder>().addComponents(
+        reasonInput,
+      ),
+    )
+
+    // Show the modal
+    await interaction.showModal(reasonModal)
+
+    const submitted = await interaction.awaitModalSubmit({
+      filter: (interaction) => interaction.user.id === interaction.user.id,
```

Comment:
this condition is always true, maybe `interaction_`?

Reply:
```suggestion
      filter: (interaction_) => interaction_.user.id === interaction.user.id,
```
like this? @dtinth

Reply:
yes krub!